tests/store/artifact/test_mlflow_artifact_repo.py

```diff
@@ -113,15 +114,24 @@ def mlflow_artifact_repo_with_host():
 def test_log_artifact(mlflow_artifact_repo, tmpdir, artifact_path):
     tmp_path = tmpdir.join("a.txt")
     tmp_path.write("0")
-    with mock.patch("requests.Session.put", return_value=MockResponse({}, 200)) as mock_put:
+    with mock.patch(
+        "mlflow.store.artifact.http_artifact_repo.http_request",
+        return_value=MockResponse({}, 200),
+    ) as mock_put:
         mlflow_artifact_repo.log_artifact(tmp_path, artifact_path)
-        paths = (artifact_path,) if artifact_path else ()
-        expected_url = posixpath.join(mlflow_artifact_repo.artifact_uri, *paths, tmp_path.basename)
+        paths = (artifact_path, tmp_path.basename) if artifact_path else (tmp_path.basename,)
         mock_put.assert_called_once_with(
-            expected_url, data=FileObjectMatcher(tmp_path, "rb"), timeout=mock.ANY
+            mlflow_artifact_repo.host_creds,
```

Comment:
Can we add a test case that configures some auth via environment variables and then asserts that these host credentials contain the expected auth configs?

Reply:
Can we do this for `HttpArtifactRepository` as well after addressing https://github.com/mlflow/mlflow/pull/5370#discussion_r805223471?

Reply:
Added a test to `test_http_artifact_repo.py`. I don't think adding the same test to `test_mlflow_artifact_repo.py` brings any benefit, since it would be testing the same functionality, but let me know what you think:)